tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs

```diff
@@ -264,12 +275,27 @@ private void ExpandLargeContentPanel()
         LargeContentPanel.Visibility = Visibility.Visible;
         MaxHeight = double.NaN;
 
-        // If they expand to ExpandedView and they're not snapped, we can use the
-        // RestoreState size & position.
+        var monitorRect = GetMonitorRectForWindow(ThisHwnd);
+        var monitor = PInvoke.MonitorFromWindow(ThisHwnd, MONITOR_FROM_FLAGS.MONITOR_DEFAULTTONEAREST);
+        PInvoke.GetScaleFactorForMonitor(monitor, out DEVICE_SCALE_FACTOR scaleFactor).ThrowOnFailure();
+        var dpiScale = (double)scaleFactor / 100;
+
+        // If they expand to ExpandedView and they're not snapped
         if (!_viewModel.IsSnapped)
         {
+            // Conversely if they're snapped, the position is determined by the snap,
+            // and we potentially adjust the size to ensure it doesn't extend beyond the screen.
+            var availableWidth = monitorRect.Width - Math.Abs(AppWindow.Position.X - monitorRect.left) - RightSideGap;
+            _restoreState.Width = (int)((double)availableWidth / dpiScale);
+
+            Width = _restoreState.Width;
+
+            var availableHeight = monitorRect.Height - Math.Abs(AppWindow.Position.Y - monitorRect.top);
+
+            _restoreState.Height = (int)((double)availableHeight / dpiScale);
+
             this.MoveAndResize(
-                _restoreState.Left, _restoreState.Top, _restoreState.Width, _restoreState.Height);
+                AppWindow.Position.X, AppWindow.Position.Y, _restoreState.Width, _restoreState.Height);
         }
         else
         {

```

Comment:
Since we don't expand when snapped anymore (medium size window), does the code in this else statement run anymore?

Reply:
I believe you're right I'll delete this and see what happens.

Reply:
Deleted this code and I haven't seen any negative repercussions so far.